server_tests/lib/util.js

```diff
@@ -27,25 +27,42 @@ var Util = function(client) {
      *
     */
     this.restoreDefaultOptions = function(optionsFile) {
+        var sessionId;
         var deferred = Q.defer();
-        fs.readFile(optionsFile, 'utf8', function(err, data) {
-            var sessionId;
-            if(err) {
-                deferred.resolve(JSON.parse(err));
-            } else {
+
+        return Q.all([fs.readFile(optionsFile, "utf8", function onChanged (err, data) {
+            if (err) {
+                JSON.parse(err);
```

Comment:
This probably should be `deferred.reject(err)`, I don't think err would be a string?